client/src/context/ProfileContext.js

```diff
@@ -0,0 +1,163 @@
+import React, { useReducer } from 'react';
+import axios from 'axios';
+
+
+//Created Context
+export const ProfileContext = React.createContext();
+
+
+//Stored action types names
+
+const types = {
+	Set_All_Profiles: "Set_All_Profiles",
+	Set_Is_Loading: "Set_Is_Loading",
+	Set_Error: "Set_Error",
+	Set_Profile: "Set_Profile",
+	Clear_Profile :"Clear_Profile",
+};
+
+
+//Stored actions in a reducer
+
+const profileReducer = (state, action) => {
+	switch (action.type) {
+	case types.Set_Is_Loading:
+		return { ...state, isLoading: true };
+	case types.Set_Error:
+		return { ...state, isLoading: false, error:action.payload };
+	case types.Set_All_Profiles:
+		return { ...state, allProfiles: action.payload, isLoading: false };
+	case types.Set_Profile:
+		return { ...state, profile: action.payload, isLoading: false };
+	case types.Clear_Profile:
+		return { ...state, profile: null, isLoading: false };
```

Comment:
You could save a few lines and reuse your `Set_Profile` action by passing `null` as the payload in your action?